hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/HttpClientsWorkloadReporter.java

```diff
@@ -53,4 +89,45 @@ int getHttp2QueueSize() {
                     .map(Queue::size)
                     .mapToInt(i -> i)
                     .sum();
+
+    private final Function<HttpClient, Integer> getHttp1ActiveConnectionsCount = httpClient ->
+            getHttp1ConnectionPool(httpClient)
+                    .map(DuplexConnectionPool::getActiveConnectionCount)
+                    .mapToInt(i -> i)
+                    .sum();
+
+    private final Function<HttpClient, Integer> getHttp1IdleConnectionsCount = httpClient ->
+            getHttp1ConnectionPool(httpClient)
+                    .map(DuplexConnectionPool::getIdleConnectionCount)
+                    .mapToInt(i -> i)
+                    .sum();
+
+    private final Function<HttpClient, Integer> getHttp2ConnectionsCount = http2Client ->
+            getHttp2ConnectionPool(http2Client)
+                    .map(MultiplexConnectionPool::getConnectionCount)
+                    .mapToInt(i -> i)
+                    .sum();
+
+    private final Function<HttpClient, Integer> getHttp2PendingConnectionsCount = http2Client ->
+            getHttp2ConnectionPool(http2Client)
+                    .map(MultiplexConnectionPool::getPendingCount)
+                    .mapToInt(i -> i)
```

Comment:
You can use `Function.identity()` to be more concise :)

Reply:
Unfortunately I can’t as mapToInt() requires ToIntFunction which is different than Function.